### PR TITLE
Seldon upgrade 1.6.0->1.14.0

### PR DESCRIPTION
**Pull request by DnPlas**
Upgrades seldon-core-operator to 1.14.0. This PR introduces the following:
* Updates manifest files to match latest release
* Upstream seldon-core image version bump
* Updates libraries
* WIP updates integration and unit tests

Note: These changes will be tested in k8s 1.21 in this PR and 1.22 in a different one.

### CODE REVIEW

**Comment by DnPlas** [comment]
These changes are working for me locally, I'm not sure 1) why the CI is not running after my last commit and 2) why we are getting `is forbidden: User "system:serviceaccount:testing:seldon-controller-manager"` kind of messages. Will investigate further.

**Comment by natalian98** [comment]
It looks that the grafana library was updated again very recently. Could you do the `charmcraft fetch-lib` to gather it?

File: src/charm.py [hunk 1]
```diff
@@ -80,26 +80,6 @@ def set_pod_spec(self, event):
...
-            "ENGINE_DEFAULT_CPU_LIMIT": config["engine-default-cpu-limit"],
-            "ENGINE_DEFAULT_CPU_REQUEST": config["engine-default-cpu-request"],
-            "ENGINE_DEFAULT_MEMORY_LIMIT": config["engine-default-memory-limit"],
-            "ENGINE_DEFAULT_MEMORY_REQUEST": config["engine-default-memory-request"],
-            "ENGINE_LOG_MESSAGES_EXTERNALLY": str(
-                config["engine-log-messages-externally"]
-            ).lower(),
-            "ENGINE_PROMETHEUS_PATH": config["engine-prometheus-path"],
-            "ENGINE_SERVER_GRPC_PORT": config["engine-server-grpc-port"],
-            "ENGINE_SERVER_PORT": config["engine-server-port"],
```
Comment (DomFleischmann) [inline src/charm.py:102]: Why are all these configuration removed? And if they aren't useful anymore, should we remove them in juju config too?

**Review by DomFleischmann — APPROVED** [review]

**Review by DomFleischmann — CHANGES_REQUESTED** [review]
Approved too soon, can you fetch the observability libs again?

**Pull request merged**